static/app/components/events/eventReplay/index.tsx

```diff
@@ -51,20 +44,35 @@ export default function EventReplay({
   }
 
   return (
-    <ErrorBoundary mini>
-      <LazyLoad
-        component={replayPreview}
-        replaySlug={`${projectSlug}:${replayId}`}
-        orgSlug={organization.slug}
-        event={event}
-        onClickOpenReplay={() => {
-          trackAnalytics('issue_details.open_replay_details_clicked', {
-            ...getAnalyticsDataForEvent(event),
-            ...getAnalyticsDataForGroup(group),
-            organization,
-          });
-        }}
-      />
-    </ErrorBoundary>
+    <EventReplaySection>
+      <ErrorBoundary mini>
+        <LazyLoad
+          component={replayPreview}
+          replaySlug={`${projectSlug}:${replayId}`}
+          orgSlug={organization.slug}
+          event={event}
+          onClickOpenReplay={() => {
+            trackAnalytics('issue_details.open_replay_details_clicked', {
+              ...getAnalyticsDataForEvent(event),
+              ...getAnalyticsDataForGroup(group),
+              organization,
+            });
+          }}
+        />
+      </ErrorBoundary>
+    </EventReplaySection>
   );
 }
+
+export default function EventReplay({projectSlug, event, group}: Props) {
+  const organization = useOrganization();
+  const hasReplaysFeature = organization.features.includes('session-replay');
+  const project = useProjectFromSlug({organization, projectSlug});
+  const isReplayRelated = projectCanLinkToReplay(project);
+
+  if (!hasReplaysFeature || !isReplayRelated) {
+    return null;
+  }
+
+  return <EventReplayContent {...{projectSlug, event, group}} />;
```

Comment:
does typescript check the props when we use spreads like this? checking for missing & extra props mainly

Reply:
You know, I never stopped to think about this but spreading any object actually doesn't error on excess props. It does still error when you have missing props though